src/server.c

```diff
@@ -3278,13 +3297,14 @@ int processCommand(client *c) {
     /* If cluster is enabled perform the cluster redirection here.
      * However we don't perform the redirection if:
      * 1) The sender of this command is our master.
-     * 2) The command has no key arguments. */
+     * 2) The command has no key or local channel arguments. */
     if (server.cluster_enabled &&
         !(c->flags & CLIENT_MASTER) &&
         !(c->flags & CLIENT_SCRIPT &&
           server.script_caller->flags & CLIENT_MASTER) &&
-        !(!c->cmd->movablekeys && c->cmd->key_specs_num == 0 &&
-          c->cmd->proc != execCommand))
+        (is_pubsubshard_command ||
+         !(!c->cmd->movablekeys && c->cmd->key_specs_num == 0 &&
+          c->cmd->proc != execCommand)))
```

Comment:
can this be reverted now? (together with the variable)

maybe a few other things can be reverted as well?